reconcile/utils/oc.py

```diff
@@ -1138,3 +1152,42 @@ def __init__(self, log_level, message):
 
     def __bool__(self):
         return False
+
+
+def validate_labels(labels) -> Iterable[str]:
+    """
+    Validate a label key/value against some rules from
+    https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
+    """
+    if labels is None:
+        return []
+
+    err = []
+    for k, v in labels.items():
+        if len(v) > 63:
+            err.append(f'Label value longer than 63 chars: {v}')
+        pattern = '^[A-Za-z0-9][A-Za-z0-9-_\\.]{0,61}[A-Za-z0-9]$'
+        if not re.match(pattern, v):
+            err.append(f'Label value is invalid: {v}')
+
+        prefix, name = '', k
+        if '/' in k:
+            split = k.split('/')
+            if len(split) > 3:
+                err.append(f'Only one "/" allowed in label keys: {k}')
+            prefix, name = split[0], split[1]
+        if len(name) > 63:
+            err.append(f'Label key name is longer than 63 chars: {name}')
+        if not re.match(pattern, name):
+            err.append(f'Label key name is invalid: {name}')
+        if prefix:
+            if len(prefix) > 253:
```

Comment:
253 must be the weirdest magic number I've ever found. :| Can you turn it into a constant? :)